linter/lints/cpcps/lint_crl_has_idp.go

```diff
@@ -69,91 +89,103 @@ func (l *crlHasIDP) Execute(c *x509.RevocationList) *lint.LintResult {
 		}
 	}
 
-	// Ensure that the DistributionPoint is a reasonable URI. To get to the URI,
-	// we have to step inside the DistributionPointName, then step inside that's
-	// FullName, and finally read the singular SEQUENCE OF GeneralName element.
-	if !idpv.PeekASN1Tag(cryptobyte_asn1.Tag(0).ContextSpecific().Constructed()) {
-		return &lint.LintResult{
-			Status:  lint.Warn,
-			Details: "IDP should contain distributionPoint",
+	if idpv.PeekASN1Tag(cryptobyte_asn1.Tag(0).ContextSpecific().Constructed()) || idpv.PeekASN1Tag(cryptobyte_asn1.Tag(1).ContextSpecific()) {
+		// If either the distributionPoint[0] field or the
+		// onlyContainsUserCerts[1] field is present, assume we're dealing with
+		// a CRL containing Subscriber Certs.
+
+		// We read this boolean as a byte and ensure the value is 0xFF because
+		// cryptobyte.ReadASN1Boolean can't handle the custom encoding rules for
```

Comment:
Good idea about the helper function. That was fun to write.

Reply:
Thanks! You can take it a step further. To act like `String.ReadASN1Int64WithTag`, `ReadASN1BooleanWithTag` should take as input a `*cryptobyte.String` (unfortunately it can't have `*cryptobyte.String` as a receiver because that's in another package, but no big deal). It should also take as input an `asn1.Tag`. It then mutates the `*String` by reading the appropriate bytes. If you do all this you can eliminate the lines like:

```
if !idpv.ReadASN1Bytes(&onlyContainsUserCertsBytes, onlyContainsUserCertsTag) {
```

Writing this out, I'm realizing that these are also _optional_ booleans. So this should really be:

```
func readOptionalASN1BooleanWithTag(source *cryptobyte.String, out *bool, outPresent *bool, tag asn1.Tag) bool)
```

Following the example of https://pkg.go.dev/golang.org/x/crypto/cryptobyte#String.ReadOptionalASN1Boolean and https://pkg.go.dev/golang.org/x/crypto/cryptobyte#String.ReadASN1Int64WithTag.